### PR TITLE
Docs migration

### PR DESCRIPTION
**Pull request by enolfc**
# Summary 

Moving all the documentation in the wiki to this repo and adding some new stuff (egicli 🚀)

### CODE REVIEW

File: source/auth.rst [hunk 1]
```diff
@@ -1,30 +1,94 @@
...
 
-The providers natively integrated with EGI Check-in are able to authenticate
-users with OAuth 2.0 tokens without using any certificates in the process by
-relying on the OpenStack Keystone `OS-FEDERATION API <https://developer.openstack.org/api-ref/identity/v3-ext/index.html#os-federation-api>`_.
+Some users roles have special consideration in VOs:
 
-.. TODO(enolfc): discovery of providers with check-in support?
+* Users with VO Manager, VO Deputy or VO Expert Role have extra pivileges in
+  the AppDB for managing the Virtual Appliances to be avaialable at every
+  provider. Check the AppDB guide on VO-wide image list management for more
```
Comment (gwarf) [inline source/auth.rst]: Please provide a link here.

File: source/auth.rst [hunk 2]
```diff
@@ -1,30 +1,94 @@
...
+prototype and validate applications. It can be used for up to 6 month by any
+new user.
+
+.. warning::
+
+  * After the 6-month long membership in the fedcloud.egi.eu VO, you will need
+    to move to a production VO, or establish a new VO.
+
+  * The resources are not guaranteed and may be removed without notice by
+    providers. **Back-up frequently to avoid loosing your work!**
```
Comment (gwarf) [inline source/auth.rst:49]: Would be nice if we could provide some pointers for this. (or we can agree it's out of our scope)

File: source/auth.rst [hunk 3]
```diff
@@ -1,30 +1,94 @@
...
+
+For joining this VO, just place an order in the `EGI Marketplace <https://marketplace.egi.eu/31-cloud-compute>`_
+and once approved you will be able to interact with the infrastructure.
+and we will invite providers from the infrastructure to support your needs.
```
Comment (gwarf) [inline source/auth.rst]: To be rephrased/merged with previous sentence (there is a period that should probably not be there)

File: source/auth.rst [hunk 4]
```diff
@@ -1,30 +1,94 @@
...
+
```
Comment (gwarf) [inline source/auth.rst]:
```suggestion
```

File: source/auth.rst [hunk 5]
```diff
@@ -35,8 +99,8 @@ The process for authentication is as follows:
 
 The sections below detail how to achieve this as a EGI Cloud user.
 
-EGI Check-in FedCloud client
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Check-in Access tokens
+^^^^^^^^^^^^^^^^^^^^^^
 
 Access tokens can obtained via several mechanisms, usually involving the use of
```
Comment (gwarf) [inline source/auth.rst]:
```suggestion
 Access tokens can be obtained via several mechanisms, usually involving the use of
```

File: source/auth.rst [hunk 6]
```diff
@@ -145,34 +167,44 @@ commands for performing actual operations in the provider:
...
-       'https://aai.egi.eu/oidc/token' | jq -r '.access_token')
-     export OS_TOKEN=$(openstack token issue -c id -f value)
+Legacy x.509 AAI
+----------------
 
+.. warning::
 
-VOMS (to be deprecated)
------------------------
+   OpenID Connect is the preferred federated identity tehcnoloy on EGI Cloud.
```
Comment (gwarf) [inline source/auth.rst]:
```suggestion
   OpenID Connect is the preferred federated identity technology on EGI Cloud.
```

File: source/containers.rst [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+You can run containers on EGI Cloud by:
+
+* Using a pre-configured image with docker (see EGI Docker VA) below;
+
+* installing docker on top of an existing VM (by following `docker installation
+  instructions <https://docs.docker.com/engine/installation/>`_; or
+
+* running a container orchestration platform that manages the applications on
+  top of a set of nodes (Kubernetes, Docker Swarm and Mesos are examples of
+  such orchestration platforms.
```
Comment (gwarf) [inline source/containers.rst]:
```suggestion
  such orchestration platforms).
```

File: source/containers.rst [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+and `docker-compose <https://docs.docker.com/compose/>`_ pre-installed.
+
+You can start that image as any other VA available from AppDB:
+
+#. Go to the `EGI Docker image entry in AppDB <https://appdb.egi.eu/store/vappliance/docker.ubuntu.16.04>`_.
+
+#. Check the identifiers of the endpotint, image and flavor you want to use
+   at the provider.
+
+#. Use a ssh-key when, so you can log into the VM once it's instantiated.
```
Comment (gwarf) [inline source/containers.rst]:
```suggestion
#. Use a ssh key when, so you can log into the VM once it's instantiated.
```

File: source/containers.rst [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+#. Once up, just ssh and start using docker as usual.
```
Comment (gwarf) [inline source/containers.rst]:
```suggestion
#. Once up, just ssh in the VM and start using docker as usual.
```

File: source/faq.rst [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+
+There is a VO available for 6 months piloting activities that any researcher in
+Europe can join. Just place an order into the `EGI Marketplace <https://marketplace.egi.eu/31-cloud-compute>`_.
+
+How can I get an OAuth2.0 token?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Authentication via CLI or API requires a valid Check-in token. The
+FedCloud Check-in client allows you to get one as needed. Check the
+authentication guide for more information.
```
Comment (gwarf) [inline source/faq.rst]: A link would be nice.

File: source/faq.rst [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+EGI Cloud endpoints use https so information passed to contextualise the VMs
+can be assumed to be safe and only readable within your VM. However, take into
+account that anyone with access to the VM may be able to access also the
+contextualisation information.
+
+The disk on my VM is full, how can I get more space?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are several ways to increase the disk space available at the VM. The
+fastest and easiest one is to use block storage, creating a new storage disk
```
Comment (gwarf) [inline source/faq.rst:168]: A link to related documentation would be nice

File: source/federation.rst [hunk 12]
```diff
@@ -0,0 +1,410 @@
...
+provided by Check-in OpenID Connect Identity provider. Support builds on
+the `AAI guide for SPs <https://wiki.egi.eu/wiki/AAI_guide_for_SPs>`_ with
+detailed configuration provided at the `EGI Cloud integration manual <https://egi-federated-cloud-integration.readthedocs.io/en/latest/openstack.html#openid-connect-support>`_.
+
+The integration relies on the OpenStack Keystone `OS-FEDERATION API <https://developer.openstack.org/api-ref/identity/v3-ext/index.html#os-federation-api>`_.
+
+Legacy VOMS / X.509 certificates
+::::::::::::::::::::::::::::::::
+
+EGI can support those users still using X.509 certificates extended
```
Comment (gwarf) [inline source/federation.rst]:
```suggestion
EGI can support users still using X.509 certificates extended
```

File: source/federation.rst [hunk 13]
```diff
@@ -0,0 +1,410 @@
...
+with VO attributes (e.g. acknowledging that the user is member of the VO)
+in a so called VOMS proxy. This VOMS proxy certificate is used in subsequent
+calls to the endpoints which map the certificate and VO information
+via specific integration modules for VOMS authentication.
+
+There are two implementations for the support of VOMS proxies:
+
+* `OpenNebula Perun integration <https://github.com/EGI-FCTF/fctf-perun>`_
```
Comment (gwarf) [inline source/federation.rst]: This shouldn't be in EGI-FCTGF, we shouldn't refer to an org described as deprecated.

File: source/federation.rst [hunk 14]
```diff
@@ -0,0 +1,410 @@
...
+* ``org.openstack.nova``,
+* ``org.openstack.swift``,
+* ``eu.egi.cloud.accounting``,
+* ``eu.egi.cloud.vm-management.occi``, and
+* ``eu.egi.cloud.vm-metadata.marketplace``.
+
+All providers **must** enter cloud service endpoints to GOCDB to enable
+integration with EGI.. 
+
+The `Cloud-info-provider <https://github.com/egi-foundation/cloud-info-provider>`_
```
Comment (gwarf) [inline source/federation.rst]:
```suggestion
The `Cloud-info-provider <https://github.com/EGI-Foundation/cloud-info-provider>`_
```

File: source/federation.rst [hunk 15]
```diff
@@ -0,0 +1,410 @@
...
++---------------------------+------------------+------+---------------------------------+---------------------------------------+
+| IPVersion                 | ``byte``         | No   | 4 or 6                          |                                       |
++---------------------------+------------------+------+---------------------------------+---------------------------------------+
+| IPCount                   | ``int(11)``      | No   | The number of IP addresses of   |                                       |
+|                           |                  |      | IPType this user currently      |                                       |
+|                           |                  |      | assigned to them                |                                       |
++---------------------------+------------------+------+---------------------------------+---------------------------------------+
+
+A JSON schema defining a valid Public IP Usage message can be found at: https://github.com/apel/apel/blob/9476bd86424f6162c3b87b6daf6b4270ceb8fea6/apel/db/__init__.py
```
Comment (gwarf) [inline source/federation.rst]: To be replaced by an RST link

File: source/federation.rst [hunk 16]
```diff
@@ -0,0 +1,410 @@
...
+Portal. The `Accounting portal <http://accounting.egi.eu/>`_ also runs SSM to
+receive these summaries and provides a web view of the accounting data received
+from the Resource Providers.
+
+Accounting Probes
+:::::::::::::::::
+
+Implementation of the extactor probes for accounting are listed below:
+
+* OpenNebula: https://github.com/EGI-FCTF/oneacct_export
```
Comment (gwarf) [inline source/federation.rst]: This repository too should be migrated to EGI-Foundation

File: source/federation.rst [hunk 17]
```diff
@@ -0,0 +1,410 @@
...
+
+Roadmap
+-------
+
+The TCB-Cloud board defines the roadmap for the technical evolution of the EGI
+Cloud. All the components are continuously maintained to:
+
+* Improve their programmability, providing complete APIs specification in
+  adequate format for facilitating the generation clients (e.g. following the
+  OpenAPI initiative) and Swagger.
```
Comment (gwarf) [inline source/federation.rst]: Shouldn't and Swagger be inside the ()?

Comment (gwarf) [reply source/federation.rst]:
```suggestion
  OpenAPI initiative and Swagger).
```

File: source/federation.rst [hunk 18]
```diff
@@ -0,0 +1,410 @@
...
+The TCB-Cloud board defines the roadmap for the technical evolution of the EGI
+Cloud. All the components are continuously maintained to:
+
+* Improve their programmability, providing complete APIs specification in
+  adequate format for facilitating the generation clients (e.g. following the
+  OpenAPI initiative) and Swagger. 
+
+* Lower the barriers to integrate and operate resource centres in the
+  federation by a) minimizing the number of components used; b) contributing
+  code to upstream distributions; and c) use only public APIs of the Cloud
```
Comment (gwarf) [inline source/federation.rst:401]:
```suggestion
  code to upstream distributions; and c) using only public APIs of the Cloud
```

File: source/solutions/htc.rst [hunk 19]
```diff
@@ -0,0 +1,388 @@
+Deploying HTC clusters
+======================
+
+This document describes how to use the Elastic Cloud Computing Cluster (EC3)
```
Comment (gwarf) [inline source/solutions/htc.rst:4]: A link to EC3 homepage would be nice.

File: source/training.rst [hunk 20]
```diff
@@ -0,0 +1,172 @@
+.. toctree::
+
+Training infrastructure
+=======================
+
+The training infrastructure is a resource pool within the EGI Federatec Cloud
```
Comment (gwarf) [inline source/training.rst]:
```suggestion
The training infrastructure is a resource pool within the EGI Federated Cloud
```

File: source/training.rst [hunk 21]
```diff
@@ -0,0 +1,172 @@
...
+infrascture providing IaaS as well ass access services (login, application
+catalogue and application management portal) for face-to-face events, online
+training courses or self-paced learning modules.
+
+The training infrastructure is integrated with Check-in allowing trainers to
+generate short-living user accounts for training participants. Such accounts
```
Comment (gwarf) [inline source/training.rst]:
```suggestion
generate short-lived user accounts for training participants. Such accounts
```

File: source/training.rst [hunk 22]
```diff
@@ -0,0 +1,172 @@
...
+`An introductory slideset and poster of the infrastructure <https://indico.egi.eu/indico/contributionDisplay.py?contribId=122&confId=2544>`_
+from EGI Community Forum 2015, Bari. (Outdated in some parts).
+
+Usage models
+------------
+
+The training infrastructure is suitable for two types of courses:
+
+* Cloud computing courses: Such courses teach students about IaaS clouds and
+  the how Virtual Appliances, Virtual Machines, block storage and other types
```
Comment (gwarf) [inline source/training.rst]:
```suggestion
  on how Virtual Appliances, Virtual Machines, block storage and other types
```

File: source/training.rst [hunk 23]
```diff
@@ -0,0 +1,172 @@
...
+The list of providers and VAs is also discoverable in the `training.egi.eu VO
+entry of AppDB <https://appdb.egi.eu/store/vo/training.egi.eu>`_. The VO is
+also described at the `EGI Operations Portal training.egi.eu VO id card <http://operations-portal.egi.eu/vo/view/voname/training.egi.eu>`_.
+
+Modules and examples
+--------------------
+
+The training infrastructure was used in July 2015 for two face-to-face
+training courses that trained programmers about the use of the EGI Cloud (the
+first operational model above). The focus of these courses were on the use of
```
Comment (gwarf) [inline source/training.rst]:
```suggestion
first operational model above). The focus of these courses was on the use of
```

File: source/vmi.rst [hunk 24]
```diff
@@ -0,0 +1,204 @@
+.. toctree::
+
+Virtual Machine Image Management
+================================
+
+Users of the EGI Cloud, create Virtual Machines (VMs) on the providers. Those
```
Comment (gwarf) [inline source/vmi.rst]:
```suggestion
Users of the EGI Cloud create Virtual Machines (VMs) on the providers. Those
```

File: source/vmi.rst [hunk 25]
```diff
@@ -0,0 +1,204 @@
...
+  <https://wiki.appdb.egi.eu/main:guides:guide_for_managing_virtual_appliance_versions_using_the_portal>`_.
+
+* VO managers select VAs to be available at the providers following the `VO-wide
+  image list management <https://wiki.appdb.egi.eu/main:guides:manage_vo-wide_image_lists>`_.
+
+Check the full list of `Cloud marketplace guides <https://wiki.appdb.egi.eu/main:guides#cloud_marketplace>`_
+and `Cloud marketplace FAQ <https://wiki.appdb.egi.eu/main:faq#cloud_marketplace>`_
+for more information about the AppDB features.
+
+
```
Comment (gwarf) [inline source/vmi.rst]:
```suggestion
```

**Review by gwarf — CHANGES_REQUESTED** [review]
What a big documentation import, thanks!
My review probably missed some points, but we will be able to adapt/fix later if needed.
Please address the comments before we can merge this.

File: source/vmops.rst [hunk 26]
```diff
@@ -0,0 +1,80 @@
...
+
+AppDB VMOps
+===========
+
+The `EGI Application Database (AppDB) <https://appdb.egi.eu/>`_ includes a web
+GUI for management of Virtual Machines (VMs) on the federated infrastrcutrure.
+
+This GUI is available for a set of `selected VOs <https://wiki.appdb.egi.eu/main:faq:which_vos_are_supported_by_the_vmops_dashboard>`_
+If your VO is not listed and you are interested in getting support, please
+contact us!
```
Comment (gwarf) [inline source/vmops.rst]: Please provide a link/mail/contact

File: source/vmops.rst [hunk 27]
```diff
@@ -0,0 +1,80 @@
...
+* Management of VMs in *topologies*, containing one or more instances of a given
+  VA.
+
+* Attachment of additional block storage to the VM instances.
+
+* Start/Stop VMs without destroying the VM (for all VMs of a topology or for
+  individual instances within a topology)
+
+* Single control of topologies across the whole federation.
+
```
Comment (gwarf) [inline source/vmops.rst:28]:
```suggestion
```

File: source/vmops.rst [hunk 28]
```diff
@@ -0,0 +1,80 @@
...
+
+Quick start
+-----------
+
```
Comment (gwarf) [inline source/vmops.rst:31]:
```suggestion
```

File: source/vmops.rst [hunk 29]
```diff
@@ -0,0 +1,80 @@
...
+
+#. Log into the `VMOps dashboard <https://dashboard.appdb.egi.eu/vmops>`_ using
+   your EGI Check-in credentials.
```
Comment (gwarf) [inline source/vmops.rst]:
```suggestion
   EGI Check-in.
```

File: source/vmops.rst [hunk 30]
```diff
@@ -0,0 +1,80 @@
...
+
+   * Adding block storage devices to the VMs
+
+   * Define contextualisation parameters (e.g. add new users, execute some
+     script)
+
+   .. image:: img/vmops_settings.png
+      :width: 900px
+      :alt: Topology settings
+
```
Comment (gwarf) [inline source/vmops.rst:66]:
```suggestion
```

File: source/auth.rst (hunk 2 above)
Comment (enolfc) [reply source/auth.rst:49]: do we have any nice pointers?

File: source/auth.rst (hunk 3 above)
Comment (enolfc) [reply source/auth.rst]: Removed dangling sentence

File: source/auth.rst (hunk 2 above)
Comment (gwarf) [reply source/auth.rst:49]: None that I can think off, but we could have a section/placeholder/draft mentioning things like using snapshots at the IaaS level and maybe something about what you used to backup the notebooks? At least we have some knowledge/experience with this.

File: source/federation.rst (hunk 16 above)
Comment (enolfc) [reply source/federation.rst]: Linked the right project

File: source/federation.rst (hunk 13 above)
Comment (enolfc) [reply source/federation.rst]: Linked to Keystorm instead

File: source/faq.rst (hunk 11 above)
Comment (enolfc) [reply source/faq.rst:168]: Fixed

File: source/faq.rst (hunk 10 above)
Comment (enolfc) [reply source/faq.rst]: Fixed

File: source/vmops.rst (hunk 26 above)
Comment (enolfc) [reply source/vmops.rst]: Fixed

File: source/auth.rst (hunk 2 above)
Comment (enolfc) [reply source/auth.rst:49]: I have created an issue #17, to be added to the solutions or storage sections. I don't have the time to deal with this now and would like to get the docs migrated asap

**Comment by enolfc** [comment]
> What a big documentation import, thanks!
> My review probably missed some points, but we will be able to adapt/fix later if needed.
> Please address the comments before we can merge this.

Everything but the backup has been dealt with. Check if that's ok and once merged we can start pointing people here from the wiki

File: source/auth.rst (hunk 2 above)
Comment (gwarf) [reply source/auth.rst:49]: Fair enough, thanks!

**Review by gwarf — APPROVED** [review]
LGTM, 👍 +1

**Review by glarocca — COMMENTED** [review]
please change:


   docker run -v /home/centos/:/tmp/ \
              -v /home/centos/ec3/templates:/root/.ec3/templates \
              -v /tmp/.ec3/clusters:/root/.ec3/clusters grycap/ec3 launch unicam_cluster \
              torque ubuntu-1604-occi-INFN-CATANIA-STACK cluster_configure configure_nfs \
              -a /tmp/auth_INFN-CATANIA-STACK.dat

**Pull request merged**